### PR TITLE
Chapter 10 - Updating, showing, and deleting users

### PR DESCRIPTION
**Pull request by EvonneNgXinYi**
With installing gem "faker", managing the users.

### CODE REVIEW

**Comment by EvonneNgXinYi** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -32,4 +60,20 @@ def load_user
     flash[:warning] = t "users.user_not_found"
     redirect_to root_path
   end
+
+  def logged_in_user
+    unless logged_in?
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb:65]: return unless logged_in?
store_location
flash[:danger] = t "users.login_remind"
redirect_to login_path

File: app/models/user.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 class User < ApplicationRecord
+  USERS_PARAMS = %i(name email password password_confirmation).freeze
+
   attr_accessor :remember_token
 
-  USERS_PARAMS = %i(name email password password_confirmation).freeze
+  scope :sort_by_name, ->{order:name}
```
Comment (ngocvt-0484) [inline app/models/user.rb]:
```suggestion
  scope :sort_by_name, ->{order(name: :asc)}
```

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (EvonneNgXinYi) [reply app/controllers/users_controller.rb:65]: Sorry to say that: After i logged in the example user, error occurred when i call user index, profile and settings. It kept at the login page. Is it ok for me to keep original one?

**Comment by EvonneNgXinYi** [comment]
ready

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,5 +1,12 @@
 class UsersController < ApplicationController
-  before_action :load_user, only: :show
+  before_action :load_user, only: %i(edit update destroy show)
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
  before_action :load_user, except: %i(new create index)
```

File: app/models/user.rb [hunk 4]
```diff
@@ -1,7 +1,9 @@
 class User < ApplicationRecord
+  USERS_PARAMS = %i(name email password password_confirmation).freeze
+
   attr_accessor :remember_token
 
-  USERS_PARAMS = %i(name email password password_confirmation).freeze
+  scope :sort_by_name, ->{order(name: :asc)}
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
  scope :sort_by_name, ->{order name: :asc}
```

**Comment by EvonneNgXinYi** [comment]
ready

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,5 +1,12 @@
 class UsersController < ApplicationController
-  before_action :load_user, only: :show
+  before_action :load_user, except: %i(new create index)
+  before_action :logged_in_user, except: %i(show new create)
+  before_action :correct_user, only: %i(edit update)
+  before_action :admin_user, only: :destroy
+
+  def index
+    @users = User.sort_by_name.page params[:page]
```
Comment (chungpv-1008) [inline app/controllers/users_controller.rb:8]: please add the 'per' argument for this method

**Review by chungpv-1008 — COMMENTED** [review]
please fix comment, then merge the pull yourself

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (EvonneNgXinYi) [reply app/controllers/users_controller.rb:8]: image
I had set the default page in kaminari_config.rb, do i still need to add on "per" argument in this file

**Pull request merged**